adapters/repos/db/migrator_shard_status_ops.go

```diff
@@ -0,0 +1,217 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package db
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	command "github.com/weaviate/weaviate/cluster/proto/api"
+	"github.com/weaviate/weaviate/cluster/types"
+	"github.com/weaviate/weaviate/entities/errorcompounder"
+	enterrors "github.com/weaviate/weaviate/entities/errors"
+	"github.com/weaviate/weaviate/entities/models"
+)
+
+func (m *Migrator) frozen(idx *Index, frozen []string, ec *errorcompounder.ErrorCompounder) {
+	if m.cluster == nil {
+		ec.Add(fmt.Errorf("no cluster exists in the migrator"))
+		return
+	}
+
+	idx.backupMutex.RLock()
+	defer idx.backupMutex.RUnlock()
+
+	eg := enterrors.NewErrorGroupWrapper(m.logger)
+	eg.SetLimit(_NUMCPU * 2)
```

Comment:
maybe we should not go over _NUMCPU value, wdyt? I have seen that in most of the places we are constraining goroutines to just `_NUMCPU` value

Reply:
well, then we have a mix because in the same file we have it `eg.SetLimit(_NUMCPU * 2)` , i don't have a hard opinion on it. for sure can adjust it

Reply:
i have gave it a thought and i think it would be beneficial to speed up download/upload 🤔  , given both are I/O-bound ops than CPU heavy ops